test/apps/iodemo/io_demo.cc

```diff
@@ -650,19 +662,26 @@ class P2pDemoCommon : public UcxContext {
             }
         }
 
-        void init(size_t data_size, void *ext_buf)
+        void init(size_t data_size, void *ext_buf, bool validate)
         {
             assert(ext_buf != NULL);
             assert(_iov.empty());
             assert(_extra_buf == NULL);
 
             _data_size = data_size;
             _extra_buf = ext_buf;
+            _validate  = validate;
         }
 
         inline Buffer &operator[](size_t i) const
         {
-            return *_iov[i];
+            if (i < _iov.size()) {
```

Comment:
maybe test `_extra_buf` ?

Reply:
since we access _iov vector here, IMO better test the access index directly vs. the vector size

Reply:
ah.. `_iov.size()` is `0` in case of dummy?

Reply:
yes, and to prevent access beyond vector end ad need to ad this check